pilot/pkg/networking/core/v1alpha3/listener_waypoint.go

```diff
@@ -112,7 +115,10 @@ func (lb *ListenerBuilder) buildHCMConnectTerminateChain(routes []*route.Route)
 	h.HttpFilters = []*hcm.HttpFilter{
 		xdsfilters.ConnectBaggageFilter,
 		xdsfilters.ConnectAuthorityFilter,
-		xdsfilters.Router,
+		xdsfilters.BuildRouterFilter(xdsfilters.RouterFilterContext{
+			StartChildSpan:       false,
+			SuppressDebugHeaders: ph.SuppressDebugHeaders,
```

Comment:
CONNECT router should probably not be covered by ProxyConfig - it's the tunnel headers, not application headers.